src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,76 @@
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [currentUser, setCurrentUser] = useState<User>();
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (!currentTodo) {
+      setIsLoading(false);
+
+      return;
+    }
+
+    getUser(currentTodo.id).then(user => {
+      setCurrentUser(user);
+      setIsLoading(false);
+    });
+  }, [currentTodo, dispatch]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading && <Loader />}
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {!isLoading && (
+        <div className="modal-card">
+          <header className="modal-card-head" data-cy="modal-header">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{currentTodo?.id}
+            </div>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => dispatch(setCurrentTodo(null))}
+            />
+          </header>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo?.title}
+            </p>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+            <p className="block" data-cy="modal-user">
+              {/* For not completed */}
+              {!currentTodo?.completed && (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              {/* For completed */}
+              {currentTodo?.completed && (
+                <strong className="has-text-success">Done</strong>
+              )}
+              {' by '}
+              <a href={`mailto:${currentUser?.email}`}>Leanne Graham</a>
```

Comment:
The name 'Leanne Graham' is hardcoded. Consider using `currentUser?.name` to dynamically display the user's name based on the fetched data.